pkg/operator/ceph/controller/spec.go

```diff
@@ -394,13 +394,48 @@ func ContainerEnvVarReference(envVarName string) string {
 }
 
 // DaemonEnvVars returns the container environment variables used by all Ceph daemons.
-func DaemonEnvVars(image string) []v1.EnvVar {
+func DaemonEnvVars(cephClusterSpec *cephv1.ClusterSpec) []v1.EnvVar {
+	networkEnv := ApplyNetworkEnv(cephClusterSpec)
+	cephDaemonsEnvVars := append(k8sutil.ClusterDaemonEnvVars(cephClusterSpec.CephVersion.Image), networkEnv...)
+
 	return append(
-		k8sutil.ClusterDaemonEnvVars(image),
+		cephDaemonsEnvVars,
 		config.StoredMonHostEnvVars()...,
 	)
 }
 
+func ApplyNetworkEnv(cephClusterSpec *cephv1.ClusterSpec) []v1.EnvVar {
+	var rookMsgr2EnvValue string
+	if cephClusterSpec.Network.Connections != nil {
+		if cephClusterSpec.Network.Connections.RequireMsgr2 {
+			rookMsgr2EnvValue = "msgr2"
+		} else {
+			rookMsgr2EnvValue = strings.Replace(rookMsgr2EnvValue, "mgr2", "", -1)
+		}
+
+		if cephClusterSpec.Network.Connections.Compression != nil && cephClusterSpec.Network.Connections.Compression.Enabled {
```

Comment:
@travisn if `encryption` or `compression` is set to true do we automatically update the cluster CR with `msgr2:true`?
Or how it does works

Reply:
we keep watch on the cephcluster cr and every CR and if setting is change, we update accordingly